src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/EmbeddingPreview.jsx

```diff
@@ -17,487 +16,261 @@ import PlotStyling from '../../filter-cells/components/PlotStyling';
+    setConfig(plots[selectedSpec].initialConfig);
+  }, [selectedSpec]);
+
+  // Quick and dirty function to massage prepared data into a good shape.
+  // This will be changed once we actually load data from Redux.
+  const generateData = (spec) => {
+    spec.data.map((s) => {
+      if (s.name === 'cellSets') {
+        s.values = [];
+
+        UMAP.forEach((cell, i) => {
+          s.values[cell.cluster_id] = {
+            name: `${cell.cluster_id}`,
+            cellSetId: cell.cluster_id,
+            cellIds: s.values[cell.cluster_id]?.cellIds ? [...s.values[cell.cluster_id].cellIds, i] : [i],
+            color: colorProvider.getColor(),
+          };
+          // cluster_id
+        });
+      }
+
+      if (s.name === 'expression') {
+        s.values = { expression: UMAP.map((cell) => cell.doubletScore || 0) };
+      }
+
+      if (s.name === 'embedding') {
+        s.values = UMAP.map((cell) => [cell.UMAP_1, cell.UMAP_2]);
+      }
     });
-  }
-
-  changePlot(val) {
-    this.updatePlotWithChanges({ plotToDraw: val });
-    if (val) {
-      this.updatePlotWithChanges({ plotTitle: 'default clusters' });
-    } else {
-      this.updatePlotWithChanges({ plotTitle: 'cell doublet score' });
+  };
+
+  const updatePlotWithChanges = (obj) => {
+    const newConfig = _.cloneDeep(config);
+    _.merge(newConfig, obj);
+    setConfig(newConfig);
+  };
+
+  const renderPlot = () => {
+    if (error) {
+      return (
+        <PlatformError
+          description={error}
+          onClick={() => { }}
+        />
+      );
     }
-  }
-
-  render() {
-    const { config } = this.state;
 
-    const data = { embeddingCat: this.generateData() };
+    if (!config) {
+      return (
+        <center>
+          <Spin size='large' />
+        </center>
+      );
+    }
 
     return (
-      <>
-        <PageHeader
-          title={`Embedding preview (${config.plotTitle})`}
-          subTitle='Powerful data exploration'
-          style={{ width: '100%', paddingRight: '0px' }}
-        />
-        <Row>
-          <Col span={15}>
-            <Vega data={data} spec={this.generateSpec()} renderer='canvas' />
-          </Col>
+      <center>
+        <Vega spec={plotSpec} renderer='canvas' />
+      </center>
+    );
+  };
 
-          <Col span={3}>
-            <Space direction='vertical'>
-              <Tooltip title='The number of dimensions used to configure the embedding is set here. This dictates the number of clusters in the Uniform Manifold Approximation and Projection (UMAP) which is taken forward to the ‘data exploration’ page.'>
-                <Button icon={<InfoCircleOutlined />} />
-              </Tooltip>
-              <img
-                alt=''
-                src={plot1Pic}
-                style={{
-                  height: '100px',
-                  width: '100px',
-                  align: 'center',
-                  padding: '8px',
-                  border: '1px solid #000',
+  if (!config) {
```

Comment:
I just realized this part of the code is a duplicate of the one above (line 147)

Reply:
Good point. I removed it from the other place, because the config now is used for other things other than rendering the plot, so halting the whole page load until config is ready is a better strategy in my opinion.